### DISCUSSION TITLE
DIY rOS - Ready to roll again

### DISCUSSION THREAD

**Discussion by rern** [rern/rAudio#216] [General]
It's been weeks for [**rOS**](https://github.com/rern/rOS) to struggle with package issues left and right. Now the waves and wind are good enough. For those who have tried and failed, please try again.

**Comment by ceejayemm** [comment]
@rern 
Is there (will there be) a 64 bit version of DIY rOS ?  I currently am running the original version on a RPi 3B+ booting from a USB flash drive (without any problems at the moment) but would be willing to give a 64 bit version a try.

Thanks

Chris

**Reply by klassikmann** [reply]
Yes, an aarch64 Image ist possible using rOS. I did it last weekend and I'm quite satisfied.

**Reply by ceejayemm** [reply]
Ok, I had missed the option to use a 64 bit ArchLinux image at the bottom of the image list.

@rern - I have now created a 64 bit bootable USB drive using the latest rOS DIY instructions. However when I transfer the USB drive to the RPi 3B+ the PI fails to boot with the error (on the console):

missing environment variable: bootfile

I have flashed the USB drive 3 or 4 times using Manjaro / Ubuntu and get the same result.

Chris

**Reply by rern** [reply]
Arch Linux Arm 64bit utilizes U-Boot which is different from native boot. rAudio 64bit must be created on SD card, which U-Boot removed and native boot installed, then transferd to a USB drive.

As SD cards get cheaper and faster, it's easier to run and maintain rAudio on an SD card.

**Reply by ceejayemm** [reply]
@rern 
Thanks for this update, I think I understand what you are saying. 

In my case I am running a RPi 3B+ in a SmartPiCase which houses the official 7in touch screen.  This case is great but it does make it difficult to get access to the SD card reader which is inside the case hence my interest in being able to boot from a USB flash drive which is accessible outside the case.

I think however that I have found a solution. Exploring your Github site I came across the rAudio-1 64bit beta pre-release image  and the revised installation instructions for USB drives in the Update Branch readme file.  I have used the latest Raspberry PI Imager (v1.5) to flash a new USB drive with the 64 bit beta image and transferred (to /BOOT) 'ssh' and 'wifi' text files along with a backup of my existing rAudio-1 installation.  All seems to have gone well except:

1. The NAS NFS mount point for my music library was restored in Sources from the backup but was not mounted. I went to the Sources page and manually remounted the source and all was well
2. The Beta image included update 202102211, when I tried to apply the 20210223 update this failed with:

wget -q https://github.com/rern/rAudio-1/raw/main/install.sh
chmod 755 install.sh
./install.sh r1 Update main 
 !  Install file download failed.

Please try again.

Everything else seems to be working.

If you plan to release and maintain the 64 bit image then I would look to  use this via a USB drive to run rAudio-1 - I might actually invest in a small RPi 4B (I presume a 2GB version would run rAudio-1 OK ?)  to provide quicker access to the USB flash drive via the USB3  sockets on this type of machine.

Chris

**Reply by ceejayemm** [reply]
Oops! Spoke too soon.  The new 64 bit build detailed above worked fine yesterday, I used it on and off through the day.  This morning however the browser screen is 'stuck' (via the RPi Touchscreen attached and in MS Edge and Google Chrome on my PC) in the following state:

![Capture_2702](https://user-images.githubusercontent.com/25418025/109383852-3493ac80-78e1-11eb-815f-9ad00fda9356.JPG)

The album shown is also not what I was last playing and in fact hadn't been used with this build at all (unless its come back via the backup I restored as part of the build ?). The image should have top and bottom bars as well as the coverart and volume control.

I can access the RPi via SSH if necessary but browser access is completely stuck.

Chris
